nanodbc/nanodbc.cpp

```diff
@@ -2924,14 +2924,18 @@ inline void result::result_impl::get_ref_impl(short column, T& result) const
                     buffer,          // TargetValuePtr
                     buffer_size,     // BufferLength
                     &ValueLenOrInd); // StrLen_or_IndPtr
-                if (ValueLenOrInd == SQL_NO_TOTAL)
-                    out.append(buffer, col.ctype_ == SQL_C_BINARY ? buffer_size : buffer_size - 1);
-                else if (ValueLenOrInd > 0)
-                    out.append(
-                        buffer,
-                        std::min<std::size_t>(
-                            ValueLenOrInd,
-                            col.ctype_ == SQL_C_BINARY ? buffer_size : buffer_size - 1));
+                if (ValueLenOrInd == SQL_NO_TOTAL || ValueLenOrInd > 0) {
+                    std::size_t append_size = col.ctype_ == SQL_C_BINARY ? buffer_size : strlen(buffer);
+                    if (append_size == 0) {
```

Comment:
A comment would be good to explain when this `append_size` may become Zero.
A test case would be even better - in fact, nanodbc does not not accept PRs without even a trivially minimal test case accompanying a fix.

I have no time to test it myself, but perhaps retrieving data like this would reproduce your problem you are trying to solve here:

```sql
SELECT CAST('A' + CHAR(0) + 'B' + CHAR(0) + 'C' + CHAR(0) + CHAR(0) AS varchar(max)) AS v
```

Reply:
Thanks for your kind reminding! I'll modify this accordingly and add test cases in a while.

Reply:
It's a little bit tricky to construct a case to determine how should interrupting zeros in text are "correctly handled" since `buffer_size` determines how the text is chunked. In each chunk, the interrupting zeros are present at different positions, making all subsequent chars after the first zero in the buffer ignored. When `buffer_size` is changed, the zero positions in each chunk are changed, so are the resulted strings appended to the result.